common/src/main/java/com/microsoft/identity/common/migration/DefaultSharedPrefsFileManagerReencrypter.java

```diff
@@ -0,0 +1,188 @@
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.migration;
+
+import androidx.annotation.NonNull;
+
+import com.microsoft.identity.common.internal.cache.ISharedPreferencesFileManager;
+import com.microsoft.identity.common.internal.controllers.TaskCompletedCallback;
+import com.microsoft.identity.common.logging.Logger;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Default implementation of {@link ISharedPrefsFileManagerReencrypter}.
+ */
+public class DefaultSharedPrefsFileManagerReencrypter implements ISharedPrefsFileManagerReencrypter {
+
+    private static final String TAG = DefaultSharedPrefsFileManagerReencrypter.class.getSimpleName();
+    private static final ExecutorService executor = Executors.newSingleThreadExecutor();
+
+    @Override
+    public IMigrationOperationResult reencrypt(@NonNull final ISharedPreferencesFileManager fileManager,
+                                               @NonNull final IStringEncrypter encrypter,
+                                               @NonNull final IStringDecrypter decrypter,
+                                               @NonNull final ReencryptionParams params) {
+        final String methodName = ":reencrypt (sync)";
+        final Map<String, String> cacheEntries = new HashMap<>(fileManager.getAll());
+        Logger.verbose(TAG + methodName,
+                "Attempting to migrate cache entries: " + cacheEntries.size());
+        final MigrationOperationResult result = new MigrationOperationResult();
+        result.setTotalRecords(cacheEntries.size());
+        final Set<String> keysMarkedForRemoval = new HashSet<>();
+        final AtomicBoolean shouldAbort = new AtomicBoolean(false);
+        final Set<String> skipKeys = new HashSet<>();
+
+        // Decrypt everything
+        applyCacheMutation(
+                cacheEntries,
+                new Callable<Map.Entry<String, String>>() {
+                    @Override
+                    public void call(final Map.Entry<String, String> entry) throws Exception {
+                        final String decryptedText = decrypter.decrypt(entry.getValue());
+                        entry.setValue(decryptedText);
+                    }
+                },
+                result,
+                params,
+                keysMarkedForRemoval,
+                skipKeys,
+                shouldAbort
+        );
+
+        // Clear any keys marked for removal...
+        clearEntriesMarkedForRemoval(fileManager, cacheEntries, keysMarkedForRemoval);
+
+        if (shouldAbort.get()) {
+            Logger.info(TAG + methodName, "Aborting after decrypt.");
+            return result;
+        }
```

Comment:
So if this is remove all on error, this will remove everything from the map and the Shared Prefs file, then not abort, but write nothing (since there will be nothing in cacheEntries and nothing will happen).  What's the difference between these options?

Reply:
I pushed an update based on your [previous comment](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1464#discussion_r669195712); I believe this is now resolved